### PR TITLE
[Block: Post comment date]: Add link setting and block supports

### LINKED ISSUE #30573: Comment Date Block
Here is an initial take on the design for the Comment Date block. This block should only be available in the Site Editor, and potentially only when editing certain templates (Index, Single, Singular, Page).

comment-date

Design feedback is politely requested for all aspects:

* Icon
* Placeholder when there is no context present
* Toolbar actions
* Inspector settings
* Name
* Description

Figma link [here](https://www.figma.com/file/AagoBClTm0AgpfYCSKQl0m/Comment-blocks?node-id=501%3A1238).

### PR DESCRIPTION
**Pull request by carolinan**
## Description

Closes https://github.com/WordPress/gutenberg/issues/30573.

## How has this been tested?

1. Add a post comment block (using an existing comment ID)
2. Add an inner post comment date block
3. Confirm that the color and typography options work
4. Confirm that the link option works. 
5. Confirm that the time format option works. -View source and check that the correct date and time is in the `datetime` attribute and in the text.

## Types of changes

Enhancement

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by SantosGuillamot** [comment]
Thanks for the Pull Request! 🙂 

@gziolo I'm afraid I don't feel comfortable yet reviewing the code, could you take a look at it, please?

File: packages/block-library/src/post-comment-date/edit.js [hunk 1]
```diff
@@ -24,6 +28,23 @@ export default function Edit( { attributes, context, setAttributes } ) {
...
+	let commentDate = (
+		<time dateTime={ dateI18n( 'c', date ) }>
+			{ dateI18n( resolvedFormat, date ) }
+		</time>
+	);
+
+	if ( isLink ) {
+		commentDate = (
+			<a
+				href="#comment-date-pseudo-link"
```
Comment (gziolo) [inline packages/block-library/src/post-comment-date/edit.js:40]: Could we use an actual link here? In PHP it's possible with `get_comment_link( $block->context['commentId'] );`. Would it work with `useEntityProp( 'root', 'comment', 'link', commentId )` or something like that?

File: packages/block-library/src/post-comment-date/index.php [hunk 2]
```diff
@@ -19,14 +19,21 @@ function render_block_core_post_comment_date( $attributes, $content, $block ) {
 	}
 
 	$wrapper_attributes = get_block_wrapper_attributes();
+	$formatted_date     = get_comment_date(
+		isset( $attributes['format'] ) ? $attributes['format'] : '',
+		$block->context['commentId']
+	);
+	$link               = get_comment_link( $block->context['commentId'] );
+
+	if ( isset( $attributes['isLink'] ) && $attributes['isLink'] ) {
```
Comment (gziolo) [inline packages/block-library/src/post-comment-date/index.php]: `empty` is a shorter versio of the same check:
```suggestion
	if ( ! empty( $attributes['isLink'] ) ) {
```

**Review by gziolo — COMMENTED** [review]
> I'm afraid I don't feel comfortable yet reviewing the code, could you take a look at it, please?

No worries @SantosGuillamot. The code looks good. I haven't tested so I would appreciate some help on that front.

**Review by gziolo — APPROVED** [review]

File: packages/block-library/src/post-comment-date/edit.js (hunk 1 above)
Comment (carolinan) [reply packages/block-library/src/post-comment-date/edit.js:40]: Do we really want it to be clickable in the editor? Is the issue where duplicate editors were opened fixed?

Comment (carolinan) [reply packages/block-library/src/post-comment-date/edit.js:40]: I mean we have used fake links because the editor would be opened inside the editor.

Comment (gziolo) [reply packages/block-library/src/post-comment-date/edit.js:40]: We use real links in RichText:
Screen Shot 2021-10-07 at 14 07 15

I wasn't aware of a limitation like that. I see more places where the same pattern is used so it's probably fine to leave it as is for now.

Comment (carolinan) [reply packages/block-library/src/post-comment-date/edit.js:40]: Like this: https://github.com/WordPress/gutenberg/issues/30647

**Pull request merged**